optaplanner-core/src/test/java/org/optaplanner/core/api/solver/SolverManagerTest.java

```diff
@@ -179,25 +175,6 @@ public void exceptionInConsumer() {
         assertThat(solverJob1.getSolverStatus()).isEqualTo(NOT_SOLVING);
     }
 
-    @Test
-    @Timeout(60)
-    public void solveGenerics() throws ExecutionException, InterruptedException {
```

Comment:
Had to remove this, as it was testing precisely the situation that we've now made impossible.

Reply:
We need to keep this, this is valid usages. Even though our polymorphism epic has some lacking feature gaps and testings for those - and it's pushed on the backlog - we don't want to reduce our support for it.

Some examples to think about, presume CatSolution extends AnimalSolution. BeerSolution does not.
- If AnimalSolution has the @PlanningSolution annotation, then it should fail fast on BeerSolution, but not AnimalSolution nor CatSolution.
- If CatSolution has the @PlanningSolution annotation, then it should fail fast on BeerSolution and AnimalSolution, but not CatSolution.

Reply:
PLANNER-829 is for when `@PlanningSolution Cat extends @PlanningSolution Animal` and `Cat` is solved with `Solver<Animal>`.
For TestdataUnannotatedExtendedSolution, it's different though, which shouldn't fail fast:
it's when `Cat extends @PlanningSolution Animal` and `Cat` is solved with `Solver<Animal>`, but Cat doesn't have a `@PlanningSolution` annotation.

Reply:
That's a valid use case. Especially if hibernate decided to CGLib proxy the Solution class.

Reply:
Let's discuss on bluejeans - this stuff is complicated and 2 heads are better than one.